example/test_problem/Hydro/CCSN/python_script/convert_progenitor_form.py

```diff
@@ -0,0 +1,151 @@
+       if ( headers[i] == 'cell Y_e'              ): TargetCols[4] = i
+       if ( headers[i] == 'cell pressure'         ): TargetCols[5] = i
+       if ( headers[i] == 'cell angular velocity' ): TargetCols[6] = i
+elif ( form == MESA ):
+   for i in range( len(headers) ):
+       if ( headers[i] == 'logR'                  ): TargetCols[0] = i
+       if ( headers[i] == 'density'               ): TargetCols[1] = i
+       if ( headers[i] == 'temperature'           ): TargetCols[2] = i
+       if ( headers[i] == 'velocity'              ): TargetCols[3] = i
+       if ( headers[i] == 'ye'                    ): TargetCols[4] = i
+       if ( headers[i] == 'pressure'              ): TargetCols[5] = i
+       if ( headers[i] == 'omega'                 ): TargetCols[6] = i
+
+
+# GAMER-formatted array
+GAMER_DATA_FORM = np.empty( (0, nvars), dtype=np.float )
+variables       = np.empty( nvars,      dtype=np.float )
+
+outer_radius = 0.0
+outer_vel    = 0.0
+for line in f:
+    if ( form == WHS  and  line[0:first_column_size].strip() == 'wind:' ): break
+    for idx, icol in enumerate(TargetCols):
+        variables[idx] = float( line.split()[icol] )
+#       convert cell outer variables to cell centered values (radius, velocity)
+        if ( form == WHS  ):
+            if ( idx == 0 ):
+                cell_centered_radius = ( outer_radius + variables[idx] ) / 2.0
+                outer_radius         = variables[idx]
+                variables[idx]       = cell_centered_radius
+            if ( idx == 3 ):
+                cell_centered_vel    = ( outer_vel + variables[idx] ) / 2.0 
+                outer_vel            = variables[idx]
+                variables[idx]       = cell_centered_vel
+#       unit conversion
+        if ( form == MESA ):
+            if ( idx == 0 ):
+                variables[idx] = 6.977e10 * np.power( 10, variables[idx] ) # unit conversion: radius
+            if ( idx == 3 ):
+                variables[idx] = 1e2 * variables[idx]                      # unit conversion: velocity
+
+    GAMER_DATA_FORM = np.vstack( [GAMER_DATA_FORM, variables] )
+
+if ( form == MESA ):
+    GAMER_DATA_FORM = np.flipud( GAMER_DATA_FORM )
+
+# add a safer line for the innermost radius
+GAMER_DATA_FORM = np.vstack( [GAMER_DATA_FORM[0], GAMER_DATA_FORM] )
+GAMER_DATA_FORM[0, 0] = 1.0
+
+
+# close file
+f.close()
+
+
+# write GAMER-formatted header
+header_GAMER = "radius".rjust(21) + "density".rjust(25) + "temperature".rjust(25) + "radial velocity".rjust(25) \
+             + "ye".rjust(25) + "pressure".rjust(25) + "omega".rjust(25) + "\n" \
+             + "[cm]".rjust(21) + "[g/cm^3]".rjust(25) + "[K]".rjust(25) + "[cm/s]".rjust(25) \
+             + "[1]".rjust(25) + "[bar]".rjust(25) + "[rad/s]".rjust(25)
+header_GAMER = f_info + header_GAMER
+
+# write data to the text file
+fout = fin + '_GAMER'
+np.savetxt( fout, GAMER_DATA_FORM, fmt='% .16e', header=header_GAMER, delimiter='  ',
```

Comment:
- Why add one space before `.16e`?
- You do not need to pass the arguments if the default values are adopted.

Reply:
- The one space before ` .16e` is for +- signs. I would like to align the lines with decimal points.
- I removed the default values.

Reply:
I prefer to use `25.16e` instead, and it is more clear than the data are in the same format as the header

Reply:
I updated this. It is `23.16e` due to the delimiter.